### PR TITLE
feat: make ray optional

### PR DESCRIPTION
**Pull request by jarulraj**
👋 Thanks for submitting a Pull Request to EvaDB!

🙌 We want to make contributing to EvaDB as easy and transparent as possible. Here are a few tips to get you started:

- 🔍 Search existing EvaDB [PRs](https://github.com/georgia-tech-db/eva/pulls) to see if a similar PR already exists.
- 🔗 Link this PR to a EvaDB [issue](https://github.com/georgia-tech-db/eva/issues) to help us understand what bug fix or feature is being implemented.
- 📈 Provide before and after profiling results to help us quantify the improvement your PR provides (if applicable).

👉 Please see our ✅ [Contributing Guide](https://evadb.readthedocs.io/en/stable/source/contribute/index.html) for more details.

### CODE REVIEW

File: evadb/executor/exchange_executor.py [hunk 1]
```diff
@@ -65,6 +63,9 @@ def build_inner_executor(self, inner_executor):
         self.inner_executor.children = [QueueReaderExecutor()]
 
     def exec(self) -> Iterator[Batch]:
+        print("RAY")
```
Comment (jiashenC) [inline evadb/executor/exchange_executor.py:66]: Extra print

File: evadb/executor/ray_utils.py [hunk 2]
```diff
@@ -26,8 +22,10 @@ class StageCompleteSignal:
     pass
 
 
-@ray.remote(num_cpus=0)
-def ray_wait_and_alert(tasks: List[ray.ObjectRef], queue: Queue):
+def ray_wait_and_alert(tasks, queue):
```
Comment (jiashenC) [inline evadb/executor/ray_utils.py:25]: Did you try running Ray enabled? I think this won't work if you remove the decorator. 

Is this a good solution? https://stackoverflow.com/questions/16192448/conditional-definition-of-a-function-in-a-module

Comment (jarulraj) [reply evadb/executor/ray_utils.py:25]: https://docs.ray.io/en/latest/ray-core/api/doc/ray.remote.html

This seems like a simpler solution: `f_with_2_gpus = f.options(num_gpus=2) `

File: evadb/evadb.yml [hunk 3]
```diff
@@ -25,7 +25,7 @@ server:
   socket_timeout: 60
 
 experimental:
-  ray: True
+  ray: False
```
Comment (xzdandy) [inline evadb/evadb.yml:28]: We need to update the circle ci test setup, after making the default to be False.  Currently we have all tests  with ray enabled except `COVERAGE`. If we want to keep this test setup, we need to modify the circleci configuration accordingly.

File: evadb/optimizer/rules/rules_manager.py [hunk 4]
```diff
@@ -115,7 +116,8 @@ def __init__(self, config: ConfigurationManager):
             LogicalVectorIndexScanToPhysical(),
         ]
 
-        ray_enabled = config.get_value("experimental", "ray")
+        # ray_enabled = config.get_value("experimental", "ray")
+        ray_enabled = is_ray_available()
```
Comment (xzdandy) [inline evadb/optimizer/rules/rules_manager.py:120]: We can not turn off ray when ray is installed. Shall we do  `config.get_value("experimental", "ray") and is_ray_available()` instead?

File: evadb/executor/ray_utils.py (hunk 2 above)
Comment (jiashenC) [reply evadb/executor/ray_utils.py:25]: I think if we specify `num_gpus` on the Ray side, this (https://github.com/georgia-tech-db/eva/pull/817) issue may occur.

File: evadb/optimizer/rules/rules_manager.py (hunk 4 above)
Comment (jarulraj) [reply evadb/optimizer/rules/rules_manager.py:120]: @gaurav274  We could probably re-enable this option. In addition to ray being installed, we can let user update this config variable to actually use `ray`. That will further reduce the likelihood of `ray` being accidentally used.

File: evadb/evadb.yml (hunk 3 above)
Comment (jarulraj) [reply evadb/evadb.yml:28]: Yes, by default, let us disable it. For each python version, we could have build with ray disabled and ray enabled (ray installed and configuration variable set to True).

**Review by gaurav274 — APPROVED** [review]

**Pull request merged**